src/models.rs

```diff
@@ -985,6 +985,47 @@ pub struct ContextDocsResult {
     pub detections: Vec<DetectionResult>,
 }
 
+/// The request format expected in the /api/v2/text/detect/generated endpoint.
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct DetectionOnGeneratedHttpRequest {
+    /// The prompt to be sent to the LLM.
+    #[serde(rename = "prompt")]
+    pub prompt: String,
+
+    /// The text generated by the LLM.
+    #[serde(rename = "generated_text")]
+    pub generated_text: String,
+
+    /// The map of detectors to be used, along with their respective parameters, e.g. thresholds.
+    #[serde(rename = "detectors")]
+    pub detectors: HashMap<String, DetectorParams>,
+}
+
+impl DetectionOnGeneratedHttpRequest {
+    /// Upfront validation of user request
+    pub fn validate(&self) -> Result<(), ValidationError> {
+        // Validate required parameters
+        if self.prompt.is_empty() {
+            return Err(ValidationError::Required("prompt".into()));
+        }
+        if self.generated_text.is_empty() {
+            return Err(ValidationError::Required("generated_text".into()));
+        }
+        if self.detectors.is_empty() {
+            return Err(ValidationError::Required("detectors".into()));
+        }
+        Ok(())
+    }
+}
+
+/// The response format of the /api/v2/text/detection/generated endpoint
+#[derive(Default, Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct DetectionOnGenerationResult {
+    /// Detection results
+    #[serde(rename = "detections")]
+    pub detections: Vec<DetectionResult>,
+}
+
```

Comment:
This structure has the same field as `ContextDocsResult`, but I coudn't think of a good name to reuse the already existing one. Any suggestions?

Reply:
Sure.